lib/types/rules/possible-errors.d.ts

```diff
@@ -251,7 +251,17 @@ export interface PossibleErrors extends Linter.RulesRecord {
      * @since 0.4.0
      * @see https://eslint.org/docs/rules/no-extra-boolean-cast
      */
-    "no-extra-boolean-cast": Linter.RuleEntry<[]>;
+    "no-extra-boolean-cast": Linter.RuleEntry<
+        [
+            Partial<{
+                /**
+                 * @since 9.3.0
+                 * @default false
+                 */
+                enforceForInnerExpressions: boolean;
+            }>,
```

Comment:
Let's include `enforceForLogicalOperands ` too because it does exist. Let's have something like below.
```js
           Partial<{
                enforceForInnerExpressions: boolean;
            }>
            
            |
        
            Partial<{
                /**
                 * @deprecated Use `enforceForInnerExpressions` instead
                 */
                enforceForLogicalOperands: boolean;
            }>
 ```

Reply:
Thanks for the tip. Unfortunately that doesn’t make the two options mutually exclusive.

However, the `never` type can be leveraged for such functionality as seen in my most recent push.